src/test/java/com/google/research/bleth/services/DatabaseServiceTest.java

```diff
@@ -0,0 +1,257 @@
+
+    @Mock
+    AbstractSimulation.Builder firstSimulationBuilder;
+
+    @Mock
+    AbstractSimulation.Builder secondSimulationBuilder;
+
+    @Mock
+    Beacon beacon;
+
+    @Mock
+    Observer observer;
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @Before
+    public void mockFirstSimulationBuilder() {
+        Mockito.when(firstSimulationBuilder.getSimulationType()).thenReturn("Stalking");
+        Mockito.when(firstSimulationBuilder.getMaxNumberOfRounds()).thenReturn(100);
+        Mockito.when(firstSimulationBuilder.getBeaconsNum()).thenReturn(2);
+        Mockito.when(firstSimulationBuilder.getObserversNum()).thenReturn(1);
+        Mockito.when(firstSimulationBuilder.getRowNum()).thenReturn(2);
+        Mockito.when(firstSimulationBuilder.getColNum()).thenReturn(2);
+        Mockito.when(firstSimulationBuilder.getBeaconMovementStrategy()).thenReturn(new RandomMovementStrategy());
+        Mockito.when(firstSimulationBuilder.getObserverMovementStrategy()).thenReturn(new StationaryMovementStrategy());
+        Mockito.when(firstSimulationBuilder.getAwakenessStrategyType()).thenReturn(AwakenessStrategyFactory.Type.FIXED);
+        Mockito.when(firstSimulationBuilder.getRadius()).thenReturn(3.5);
+        Mockito.when(firstSimulationBuilder.getAwakenessCycle()).thenReturn(5);
+        Mockito.when(firstSimulationBuilder.getAwakenessDuration()).thenReturn(1);
+    }
+
+    @Before
+    public void mockSecondSimulationBuilder() {
+        Mockito.when(secondSimulationBuilder.getSimulationType()).thenReturn("Stalking");
+        Mockito.when(secondSimulationBuilder.getMaxNumberOfRounds()).thenReturn(100);
+        Mockito.when(secondSimulationBuilder.getBeaconsNum()).thenReturn(2);
+        Mockito.when(secondSimulationBuilder.getObserversNum()).thenReturn(1);
+        Mockito.when(secondSimulationBuilder.getRowNum()).thenReturn(2);
+        Mockito.when(secondSimulationBuilder.getColNum()).thenReturn(2);
+        Mockito.when(secondSimulationBuilder.getBeaconMovementStrategy()).thenReturn(new RandomMovementStrategy());
+        Mockito.when(secondSimulationBuilder.getObserverMovementStrategy()).thenReturn(new StationaryMovementStrategy());
+        Mockito.when(secondSimulationBuilder.getAwakenessStrategyType()).thenReturn(AwakenessStrategyFactory.Type.FIXED);
+        Mockito.when(secondSimulationBuilder.getRadius()).thenReturn(3.5);
+        Mockito.when(secondSimulationBuilder.getAwakenessCycle()).thenReturn(5);
+        Mockito.when(secondSimulationBuilder.getAwakenessDuration()).thenReturn(1);
+    }
+
+    @Before
+    public void mockAgents() {
+        Mockito.when(beacon.getId()).thenReturn(0);
+        Mockito.when(beacon.getType()).thenReturn("Beacon");
+        Mockito.when(observer.getId()).thenReturn(0);
+        Mockito.when(observer.getType()).thenReturn("Observer");
+    }
+
+    @Test
+    public void writeNonEmptyRealBoardThenReadRealBoard_shouldGetExpectedJson() {
```

Comment:
Consider mentioning that you should get this Json when you read from db - the name only mentions writing to it.

Reply:
writeNonEmptyRealBoard**ThenRead**RealBoard_shouldGetExpectedJson